File: src/day1.py
in_list = []
seen = set()

with open('../data/day1.txt') as f:
    for line in f:
        in_list.append(int(line))


print("Part 1")
for item in in_list:
    for elem in seen:
        if item + elem == 2020:
            print(str(item * elem))
    seen.add(item)

print("Part 2")
for item in in_list:
    for i in seen:
        for j in seen:
            if item + i + j == 2020: 
                print(str(item * i * j))
    seen.add(item)

File: src/Day1.hs
module Day1 where

import System.Environment
import System.IO
import Data.List.Split
import Data.Set (member, fromList)
import Text.Printf

run :: IO ()
run = do
    input <- map read.lines <$> readFile "data/Day1.txt"
    printf "Part 1:\t%d\n" $ p1 input
    printf "Part 2:\t%d\n" $ p2 input

p1 :: [Int] -> Int
p1 input = head [x * (2020 - x) | x <- input, member (2020 - x) set]
    where
        set = fromList input

p2 :: [Int] -> Int
p2 input = head [ x * y * (2020 - (x+y)) | x <- input, y <- input, member (2020 - (x+y)) set]
    where 
        set = fromList input
